source/hpack.c

```diff
@@ -47,71 +49,82 @@ static uint8_t s_masked_right_bits_u8(uint8_t num_masked_bits) {
     return UINT8_MAX >> cut_bits;
 }
 
-size_t aws_hpack_get_encoded_length_integer(uint64_t integer, uint8_t prefix_size) {
-    const uint8_t prefix_mask = s_masked_right_bits_u8(prefix_size);
-
-    if (integer < prefix_mask) {
-        /* If the integer fits inside the specified number of bits but won't be all 1's, then that's all she wrote */
+static int s_append_u8_dynamic(struct aws_byte_buf *output, uint8_t u8) {
+    struct aws_byte_cursor cursor = aws_byte_cursor_from_array(&u8, 1);
+    return aws_byte_buf_append_dynamic(output, &cursor);
+}
 
-        return 1;
-    } else {
-        integer -= prefix_mask;
+/* If buffer isn't big enough, grow it intelligently */
+static int s_ensure_space(struct aws_byte_buf *output, size_t required_space) {
+    size_t available_space = output->capacity - output->len;
+    if (required_space <= available_space) {
+        return AWS_OP_SUCCESS;
+    }
 
-        size_t num_bytes = 1;
-        do {
-            ++num_bytes;
-            integer >>= 7;
-        } while (integer);
-        return num_bytes;
+    /* Capacity must grow to at least this size */
+    size_t required_capacity;
+    if (aws_add_size_checked(output->len, required_space, &required_capacity)) {
+        return AWS_OP_ERR;
     }
+
+    /* Prefer to double capacity, but if that's not enough grow to exactly required_capacity */
+    size_t double_capacity = aws_add_size_saturating(output->capacity, output->capacity);
+    size_t reserve = required_capacity > double_capacity ? required_capacity : double_capacity;
```

Comment:
aws_size_max()?

Reply:
YES
written before I added the min/max functions

I've only had them a few days and I love them to death